test/common/network/connection_impl_test.cc

```diff
@@ -797,6 +797,38 @@ TEST_P(ConnectionImplTest, BasicWrite) {
   disconnect(true);
 }
 
+// Validate dynamic socket options
+TEST_P(ConnectionImplTest, DynamicSocketOptions) {
+  setUpBasicConnection();
+  connect();
+
+  auto dynamic_sockopt_ptr = client_connection_->getDynamicSocketOptionsPtr();
+  {
+    // Test set and get socket receive buffer size
+    uint32_t set_recv_buf_size = 81920;
+    uint32_t get_recv_buf_size = 0;
+    dynamic_sockopt_ptr->setSocketRecvBufferSize(set_recv_buf_size);
+    dynamic_sockopt_ptr->getSocketRecvBufferSize(get_recv_buf_size);
+#ifdef __APPLE__
+    // 'set_recv_buf_size' is equal to 'get_recv_buf_size' on macOS
+    EXPECT_EQ(get_recv_buf_size, set_recv_buf_size);
```

Comment:
unit test error handling where possible.  Setting to negative value should work for setters. For getters it might be problematic without an Os Singleton override.

Reply:
Since I will modify API to return SyscallIntResult; I will see if I can set negative values.

Reply:
Alyssa(@alyssawilk ), I experimented a bit. What I realized is if we try to set a value lesser than the default minimum to, the setsokopt API just passes with returning rc 0; however setting the minimum allowed buffer size.

The other way to reproduce rc != 0 is not initializing the socket option length. This however only fails on the CI not on my machine. So, I am going ahead with updating the PR waiting on follow up for this issue.